datajunction-server/datajunction_server/api/metrics.py

```diff
@@ -103,31 +102,28 @@ async def get_common_dimensions(
     """
     Return common dimensions for a set of metrics.
     """
-    metric_nodes = []
     errors = []
-    for node_name in metric:
-        statement = (
-            select(Node)
-            .where(Node.name == node_name)
-            .where(is_(Node.deactivated_at, None))
-        )
-        try:
-            node = session.exec(statement).one()
-            if node.type != NodeType.METRIC:
-                errors.append(
-                    DJError(
-                        message=f"Not a metric node: {node_name}",
-                        code=ErrorCode.NODE_TYPE_ERROR,
-                    ),
-                )
-            metric_nodes.append(node)
-        except NoResultFound:
+    statement = (
+        select(Node)
+        .where(Node.name.in_(metric))  # type: ignore  # pylint: disable=no-member
+        .where(is_(Node.deactivated_at, None))
+    )
+    metric_nodes = session.exec(statement).all()
+    for node in metric_nodes:
```

Comment:
Is there a way to avoid this loop-check if we add to the query that we only want Metric nodes?
Ah... I guess this is a way to not fail-quietly and instead tell the user that they gave us a bad metric name.

Reply:
Yeah, I think the idea here was tell the user about all the nodes they supplied that aren't metrics.